core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -145,10 +146,14 @@ public void process(final Archive<?> applicationArchive, final TestClass testCla
 
         LibraryContainer<?> entryPointContainer = libraryContainer;
 
-        if (applicationArchive instanceof EnterpriseArchive) {
-            for (ArchivePath applicationKey : applicationArchive.getContent().keySet()) {
-                if (applicationKey.get().endsWith(".war")) {
-                    entryPointContainer = applicationArchive.getAsType(WebArchive.class, applicationKey.get());
+        if (!archiveContains(applicationArchive, testClass.getJavaClass())) {
+            for (Node node : applicationArchive.getContent().values()) {

```

Comment:
think you can use archive.getContext(warFilter) to make it easier to understand, wdyt?